BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/ReflectionCredentialEnvelope.cs

```diff
@@ -0,0 +1,216 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Authentication
+{
+    using System;
+    using System.Linq.Expressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// This is an envelope for an instance of Azure.Core.TokenCredential.
+    /// This class uses reflection to interact with the Azure.Core library.
+    /// </summary>
+    /// <remarks>
+    /// Our SDK currently targets net452, net46, and netstandard2.0.
+    /// Azure.Core.TokenCredential is only available for netstandard2.0.
+    /// </remarks>
+    internal class ReflectionCredentialEnvelope
+    {
+        private readonly object tokenCredential;
+        private readonly object tokenRequestContext;
+
+        /// <summary>
+        /// Create an instance of <see cref="ReflectionCredentialEnvelope"/>.
+        /// </summary>
+        /// <param name="tokenCredential">An instance of Azure.Core.TokenCredential.</param>
+        public ReflectionCredentialEnvelope(object tokenCredential)
+        {
+            this.tokenCredential = tokenCredential ?? throw new ArgumentNullException(nameof(tokenCredential));
+
+            if (tokenCredential.GetType().IsSubclassOf(Type.GetType("Azure.Core.TokenCredential, Azure.Core")))
+            {
+                this.tokenRequestContext = AzureCore.MakeTokenRequestContext(scopes: CredentialConstants.GetScopes());
+            }
+            else
+            {
+                throw new ArgumentException($"The provided {nameof(tokenCredential)} must inherit Azure.Core.TokenCredential", nameof(tokenCredential));
+            }
+        }
+
+        /// <summary>
+        /// Gets the TokenCredential object held by this class.
+        /// </summary>
+        public object Credential => this.tokenCredential;
+
+        /// <summary>
+        /// Gets an Azure.Core.AccessToken.
+        /// </summary>
+        /// <param name="cancellationToken">The System.Threading.CancellationToken to use.</param>
+        /// <returns>A valid Azure.Core.AccessToken.</returns>
+        public string GetToken(CancellationToken cancellationToken = default)
```

Comment:
is it possible that this can throw an exception?

Reply:
Yes. I was leaning towards handling this insider the calling class. 
Because being unable to get a token will require behavior changes in the TelemetryChannel and QuickPulse.

Alternatively, I could handle it here and return a NULL.